specs/activity-api.yaml

```diff
@@ -200,48 +166,40 @@ paths:
             type: string
         - name: commandIndex
           description: >
-            Wait until command with the specified index finishes. Must be accompanied by a valid "timeout" query parameter.
+            Wait until command with the specified index finishes. Must be accompanied by a valid "pollTimeout" query parameter.
           required: false
           in: query
           schema:
             type: number
             format: integer
             minimum: 0
-        - $ref: 'common.yaml#/parameters/timeout'
+        - $ref: 'common.yaml#/parameters/afterTimestamp'
+        - $ref: 'common.yaml#/parameters/pollTimeout'
       responses:
         200:
           description: Success
           content:
-            application/json:
+            application/json: # standard JSON response
               schema:
                 type: array
                 items:
                   $ref: '#/components/schemas/ExeScriptCommandResult'
-        400:
-          description: Bad Request
-          content:
-            application/json:
+            text/event-stream: # an EventSource implementation (as per https://www.w3.org/TR/eventsource/)
               schema:
-                $ref: 'common.yaml#/schemas/ErrorMessage'
+                type: array
+                items:
+                  $ref: '#/components/schemas/ExeScriptCommandResult'
```

Comment:
There are several issues here:
- it's not possible to correctly specify an endpoint of Server-Sent Events with open api
- the event stream does not return an array of json objects; server-sent events have a defined format (e.g. `data: ` prefixes)
- data contains `RuntimeEvent` objects https://github.com/golemfactory/ya-client/blob/f9b7462fc80982ccccca3e0b2c9b79c9e488f229/model/src/activity/runtime_event.rs#L6-L11
 in JSON format, i.e.

```
event: runtime
data: { kind: "started", ... }

event: runtime
data: { kind: "stdout", ... }

event: runtime
data: { kind: "stdout", ... }

event: runtime
data: { kind: "finished", ... }
```

Reply:
Right, thanks for pointing at the data structure being returned in an event. 
My point here is to at least specify the response structure, even though as you said - the OpenAPI has no way of describing the "EventSource" contract. So my proposal is for me to specify the RuntimeEvent (in a similar way to the Command structure as specified in the github.com/golemfactory/ya-client/pull/53/files#diff-7325c477bed8a73a562567672156e19627dca89c6d7acb96b63d0fd8876d1870R71) and mention it here as "response structure" in the EventSource. Does that make sense?

PS: I'm now also thinking to pull back the structs defined in the exe-script.yaml into activity-api.yaml, and refer to them directly. How about that?

Reply:
> Right, thanks for pointing at the data structure being returned in an event.
> My point here is to at least specify the response structure, even though as you said - the OpenAPI has no way of describing the "EventSource" contract. So my proposal is for me to specify the RuntimeEvent (in a similar way to the Command structure as specified in the github.com/[/pull/53](https://github.com/golemfactory/ya-client/pull/53)/files#diff-7325c477bed8a73a562567672156e19627dca89c6d7acb96b63d0fd8876d1870R71) and mention it here as "response structure" in the EventSource. Does that make sense?
> 
> So my proposal is for me to specify the RuntimeEvent (in a similar way to the Command structure as specified in the github.com//pull/53/files#diff-7325c477bed8a73a562567672156e19627dca89c6d7acb96b63d0fd8876d1870R71) and mention it here as "response structure" in the EventSource. Does that make sense?

Describing `RuntimeEvent` is a good idea. My point was that the `text/event-stream` section is useless to code generators and unobvious to humans due to Open API spec limitations.